### DISCUSSION TITLE
Modify $queryString on mount

### DISCUSSION THREAD

**Discussion by stuartcusackie** [livewire/livewire#2553] [1. Help]
I have an unusual case where I would ideally like to modify the queryString in the mount hook because the values that should be excluded are not known until I run some queries. The below code has no effect, even though the format seems to be correct.

```
public function mount()
 {
        $users= User::all();

        $this->queryString['yearFrom'] = ['except' => strval($users->min('year')];
        $this->queryString['yearTo'] = ['except' => strval($users->max('year')];
}
```

To elaborate, I have a couple of select fields for selecting Min and Max Years. They are generated using for loops e..g.:

```
@for($i = $minYear; $i <= $maxYear; $i++)
<option value="{{ $i }}">{{ $i }}</option>
@endfor

```

What I'm hoping to achieve is to exclude the min and max values from the query string, because they are the defaults.

**Comment by austenc** [comment]
I also have a need for something like this. I tried using the `queryString()` method, but it seems to run before `mount()` does, so I'm not sure how to handle it. 

Edit: I know this isn't much of an answer, but perhaps there is a way to mutate it after the component is mounted. Will report back with what I find!

**Reply by joshhanley** [reply]
@austenc yeah it happens before mount. It actually listens for `component.hydrate.initial` event which is fired before mount.

The relevant code is in https://github.com/livewire/livewire/blob/95acda129edc5c96e9be62f430ddc27d31ab4cea/src/RenameMe/SupportBrowserHistory.php#L23-L41

Maybe you could a manual work around in mount?
